src/telliot_feeds/reporters/reporter360.py

```diff
@@ -0,0 +1,181 @@
+        self.last_submission_timestamp = 0
+        self.expected_profit = expected_profit
+        self.transaction_type = transaction_type
+        self.gas_limit = gas_limit
+        self.max_fee = max_fee
+        self.wait_period = wait_period
+        self.priority_fee = priority_fee
+        self.legacy_gas_price = legacy_gas_price
+        self.gas_price_speed = gas_price_speed
+        self.autopaytip = 0
+        self.staked_amount: Optional[float] = None
+        self.staker_info: Optional[StakerInfo] = None
+        self.allowed_stake_amount = 0
+
+    async def ensure_staked(self) -> Tuple[bool, ResponseStatus]:
+        # get oracle required stake amount
+        stake, status = await self.oracle.read("getStakedAmount")
+
+        if (not status.ok) or (stake is None):
+            msg = "Unable to read current stake amount"
+            return False, error_status(msg, log=logger.info)
+
+        # get accounts current stake total
+        staker_info, status = await self.oracle.read(
+            "getStakerInfo",
+            _stakerAddress=self.acct_addr,
+        )
+
+        if (not status.ok) or (staker_info is None):
+            msg = "Unable to read reporters staker info"
+            return False, error_status(msg, log=logger.info)
+
+        if self.stake is None:
+            self.stake = stake
+        else:
+            msg = "Stake amount has changed due to TRB price change."
+            logger.info(msg)
+
+        if self.staker_info is None:
+            self.staker_info = StakerInfo(*staker_info)
+        elif self.staker_info.stake_balance > staker_info[1]:
+            # update balance after depositing more stake
+            self.staker_info.stake_balance = staker_info[1]
+            logger.info("your staked balance has decreased and account might be in dispute")
+
+        account_staked_bal = self.staker_info.stake_balance
+
+        if self.stake > account_staked_bal:
+            msg = (
+                "Current stake is low. "
+                # Add more TRB to continue reporting
+                + "Approving & depositing stake."
+            )
+            logger.info(msg)
+
+            gas_price_gwei = await self.fetch_gas_price()
+            if gas_price_gwei is None:
+                return False, error_status("Unable to fetch matic gas price for staking", log=logger.info)
+
+            # diff in required stake and account staked balance
+            stake_diff = self.stake - account_staked_bal
+
+            # check TRB wallet balance!
+            wallet_balance = self.token.read("balanceOf", account=self.acct_addr)
+            logger.info(f"Current wallet TRB balance: {wallet_balance}")
+
+            if stake_diff > wallet_balance:
+                msg = "Not enough TRB in the account to cover the stake"
+                return False, error_status(msg, log=logger.warning)
+
+            txn_kwargs = {"gas_limit": 300000, "legacy_gas_price": gas_price_gwei, "_amount": stake_diff}
```

Comment:
Does the gas need to be 300k? Aren't submitValue txs like 250-260k? I would assume calling an approve would be less but idk

Reply:
I think setting gas limit a little higher helps be on the safe side cause if its low then it might try to execute the transaction but fail with the gas ending up being lost.  I could be wrong about but I think thats how it works.

Reply:
Sounds good